test/e2e/registration_service_test.go

```diff
@@ -39,15 +44,438 @@ func (s *registrationServiceTestSuite) TestLandingPageReachable() {
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Verify JSON response.
+		alive := mp["alive"]
+		require.IsType(s.T(), alive, true)
+		require.True(s.T(), alive.(bool))
+
+		environment := mp["environment"]
+		require.IsType(s.T(), environment, "")
+		require.Equal(s.T(), "e2e-tests", environment.(string))
+
+		revision := mp["revision"]
+		require.NotNil(s.T(), revision)
+
+		buildTime := mp["buildTime"]
+		require.NotNil(s.T(), buildTime)
+
+		startTime := mp["startTime"]
+		require.NotNil(s.T(), startTime)
+	})
+}
+
+func (s *registrationServiceTestSuite) TestAuthConfig() {
+	s.Run("verify_authconfig", func() {
+		// Call authconfig endpoint.
+		req, err := http.NewRequest("GET", s.route+"/api/v1/authconfig", nil)
+		require.NoError(s.T(), err)
+		client := getClient()
+
+		resp, err := client.Do(req)
+		require.NoError(s.T(), err)
+		assert.Equal(s.T(), http.StatusOK, resp.StatusCode)
+
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		require.NoError(s.T(), err)
+		require.NotNil(s.T(), body)
+
+		mp := make(map[string]interface{})
+		err = json.Unmarshal([]byte(body), &mp)
+		require.NoError(s.T(), err)
+
+		// Verify JSON response.
+		authClientLibraryURL := mp["auth-client-library-url"]
+		require.IsType(s.T(), authClientLibraryURL, "")
+		require.Equal(s.T(), authClientLibraryURL.(string), "https://sso.prod-preview.openshift.io/auth/js/keycloak.js")
+
+		authClientConfig := mp["auth-client-config"]
+		require.IsType(s.T(), authClientConfig, "")
+		mp1 := make(map[string]interface{})
+		err = json.Unmarshal([]byte(authClientConfig.(string)), &mp1)
+		require.NoError(s.T(), err)
+
+		realm := mp1["realm"]
+		require.IsType(s.T(), realm, "")
+		require.Equal(s.T(), realm.(string), "toolchain-public")
+
+		authServerURL := mp1["auth-server-url"]
+		require.IsType(s.T(), authServerURL, "")
+		require.Equal(s.T(), authServerURL.(string), "https://sso.prod-preview.openshift.io/auth")
+
+		sslRequired := mp1["ssl-required"]
+		require.IsType(s.T(), sslRequired, "")
+		require.Equal(s.T(), sslRequired.(string), "none")
+
+		resource := mp1["resource"]
+		require.IsType(s.T(), resource, "")
+		require.Equal(s.T(), resource.(string), "crt")
+
+		publicClient := mp1["public-client"]
+		require.IsType(s.T(), publicClient, true)
+		require.True(s.T(), publicClient.(bool))
+	})
+}
+
+func (s *registrationServiceTestSuite) TestSignup() {
+	// Get valid generated token for e2e tests. IAT claim is overriden
+	// to avoid token used before issued error.
+	identity0 := authsupport.NewIdentity()
+	emailClaim0 := authsupport.WithEmailClaim(uuid.NewV4().String() + "@email.tld")
+	iatClaim0 := authsupport.WithIATClaim(time.Now().Add(-60 * time.Second))
+	token0, err := authsupport.GenerateSignedE2ETestToken(*identity0, emailClaim0, iatClaim0)
+	require.NoError(s.T(), err)
+
+	s.Run("verify_post_signup_error_no_token", func() {
```

Comment:
It's minor but just in case you didn't know...
You can use `"verify post signup error no token"` in `s.Run()`. No need to use `_`

Reply:
sure.. it's just a string..